### PR TITLE
qa/suites/rbd/iscsi: enable all supported container hosts

### PR DESCRIPTION
**Pull request by idryomov**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by idryomov** [comment]
Depends on https://github.com/ceph/ceph-cm-ansible/pull/759

**Comment by idryomov** [comment]
jenkins test api

**Comment by idryomov** [comment]
jenkins test windows

**Comment by idryomov** [comment]
jenkins test windows

**Comment by idryomov** [comment]
@petrutlucian94 These Windows tests are failing consistently:
```
ceph_test_dokan.DokanTests.test_mount
ceph_test_dokan.DokanTests.test_mount_read_only
ceph_test_dokan.DokanTests.test_delete_on_close
ceph_test_dokan.DokanTests.test_io
ceph_test_dokan.DokanTests.test_subfolders
ceph_test_dokan.DokanTests.test_find_files
ceph_test_dokan.DokanTests.test_move_file
ceph_test_dokan.DokanTests.test_max_path
ceph_test_dokan.DokanTests.test_set_eof
ceph_test_dokan.DokanTests.test_set_alloc_size
ceph_test_dokan.DokanTests.test_file_type
ceph_test_dokan.DokanTests.test_volume_info
ceph_test_dokan.DokanTests.test_get_free_space
ceph_test_dokan.DokanTests.test_file_timestamp
ceph_test_dokan.DokanTests.test_delete_disposition
ceph_test_dokan.DokanTests.test_create_dispositions
ceph_test_dokan.DokanTests.test_case_sensitive
ceph_test_dokan.DokanTests.test_case_insensitive_force_lower
ceph_test_dokan.DokanTests.test_case_insensitive_force_upper
ceph_test_libcephfs_access
ceph_test_libcephfs_snapdiff
ceph_test_libcephfs_vxattr
ceph_test_libcephfs_suidsgid
ceph_test_libcephfs_lazyio
ceph_test_libcephfs_newops
ceph_test_libcephfs
```
It looks like most of them are timing out.  Could you please take a look?

**Comment by petrutlucian94** [comment]
@idryomov those are legitimate test failures, a PR https://github.com/ceph/ceph/pull/59936#issuecomment-2379005576 was forcefully merged and broke MDS (not related to Windows). It's just that the Windows build runs the libcephfs tests, the "make check" job doesn't.

We can fix it by reverting that PR.

**Comment by idryomov** [comment]
jenkins test windows

File: qa/suites/rbd/iscsi/base/install.yaml [hunk 1]
```diff
@@ -9,6 +9,10 @@ tasks:
...
     - ceph orch device ls
 - install:
-    extra_packages:
+    extra_system_packages:
+      deb:
+      - open-iscsi
+      - multipath-tools
+      rpm:
       - iscsi-initiator-utils
       - device-mapper-multipath
```
Comment (pkalever) [inline qa/suites/rbd/iscsi/base/install.yaml:18]: device-mapper-multipath is the package which has `mpathconf` tool, which is explicitly mentioned as a requirement for `rpm`  and not for `deb` package manger systems, wondering if the same package exists on Ubuntu. (Excuse me, I don't have a ubuntu VM)

**Comment by pkalever** [comment]
except above one concern/comment, rest LGTM.

**Comment by idryomov** [comment]
Ubuntu:

https://pulpito.ceph.com/dis-2024-10-01_13:26:53-rbd:iscsi-main-distro-default-smithi/

CentOS Stream:

https://pulpito.ceph.com/dis-2024-10-01_07:57:17-rbd:iscsi-main-distro-default-smithi/

**Comment by ktdreyer** [comment]
`podman` is not available on Ubuntu? Is there a ticket that tracks making it available?

**Comment by idryomov** [comment]
> `podman` is not available on Ubuntu?

IIRC it's available on Ubuntu 22.04, but not on Ubuntu 20.04.  In any case it's not installed by default, so apart from covering the Ubuntu kernel, I wanted to cover docker as well. 

Side note: we actually had some podman-specific issues with iSCSI in the past, see https://github.com/ceph/ceph/pull/42242 and https://github.com/ceph/ceph/pull/49824 for example.

**Comment by ktdreyer** [comment]
Thanks Ilya for explaining that issue with 20.04, and for explaining that it's not installed by default. Sounds like we'll need to make sure it's installed with the upcoming 24.04 (https://tracker.ceph.com/issues/66914).

Those PR links are helpful too. Thanks for sharing those.

**Pull request merged**